src/components/ResultTable.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles, MuiThemeProvider } from '@material-ui/core/styles';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Paper from '@material-ui/core/Paper';
+import '../style/Table.css';
+
+const CustomTableCell = withStyles => ({
+
+    head: {
+      paddingLeft: 10,
+      textAlign: 'left',
+      fontWeight: 'normal',
+      color: '#b9b9b9'
+    },
+    body: {
+      paddingTop: 10,
+      paddingLeft: 10,
+      paddingBottom: 10,
+      border: '1 solid #ddd',
+      borderLeft: 'none',
+      borderRight: 'none',
+      borderTop: 'none'
+    }
+  }(TableCell));
+
+const styles = {
+  root: {
+    width: '100%',
+    overflowX: 'auto'
+  },
+  row: {
+    '&:nth-of-type(odd)': {
+      backgroundColor: '#fafafa'
+    }
+  }
+};
+
+let data = [];
+
+const compare = (a, b) => {
+  if (a.time < b.time) {
+    return -1;
+  } else if (a.time > b.time) {
+    return 1;
+  } else {
+    return 0;
+  }
+};
+
+const getStationName = (shortCode, stations) => {
+  for (let key in stations) {
+    if (stations[key] === shortCode) {
+      return key;
+    }
+  }
+};
+
+const parseData = (trains, stations, currentStation, arrival) => {
+  data = [];
+  let isArrival = arrival ? 'ARRIVAL' : 'DEPARTURE';
+  let currentTime = new Date();
+
+  for (let i = 0; i < trains.length; i++) {
```

Comment:
Tätä vois lähtee refaktoroimaa sillee et eka muuttais for:it mapeiks:

```
const parseData = (trains, stations, currentStation, arrival) => {
  data = [];
  const isArrival = arrival ? 'ARRIVAL' : 'DEPARTURE';
  const currentTime = new Date();

  trains.map(train => {
   // early returnit on pop
    if (
      train.trainCategory !== 'Long-distance' &&
      train.trainCategory !== 'Commuter'
    ) {
      return;
    }

    train.timeTableRows.map(timeTableRow => {
      let scheduledTime = new Date(timeTableRow.scheduledTime);
      if (
        timeTableRow.stationShortCode === stations[currentStation] &&
        timeTableRow.type === isArrival &&
        currentTime < scheduledTime
      ) {
        let trainObject = {};
        if (train.trainCategory === 'Long-distance') {
          trainObject['number'] = train.trainType + ' ' + train.trainNumber;
        } else {
          trainObject['number'] =
            train.commuterLineID + ' ' + train.trainNumber;
        }

        trainObject['type'] = train.trainType;
        trainObject['origin'] = getStationName(
          train.timeTableRows[0].stationShortCode,
          stations,
        );
        trainObject['destination'] = getStationName(
          train.timeTableRows[train.timeTableRows.length - 1].stationShortCode,
          stations,
        );
        trainObject['time'] = scheduledTime;
        trainObject['cancelled'] = timeTableRow.cancelled;
        trainObject['estimatedTime'] =
          timeTableRow.liveEstimateTime === undefined
            ? ''
            : new Date(timeTableRow.liveEstimateTime);

        data.push(trainObject);
      }
    });
  });
};
```

sit lähtisin tekee tota sillee et sen sijaa et tos sisemmäs map:issa tehää tommone trainObject objekti, nii lopussa vaa rakentais sen objektin ja pushais sen suoraa

tyylii 

```
const parseData = (trains, stations, currentStation, arrival) => {
  const isArrival = arrival ? 'ARRIVAL' : 'DEPARTURE';
  const currentTime = new Date();

  trains.map(train => {
    if (
      train.trainCategory !== 'Long-distance' &&
      train.trainCategory !== 'Commuter'
    ) {
      return;
    }

    train.timeTableRows.map(timeTableRow => {
      const scheduledTime = new Date(timeTableRow.scheduledTime);

      if (
        timeTableRow.stationShortCode === stations[currentStation] &&
        timeTableRow.type === isArrival &&
        currentTime < scheduledTime
      ) {
        let number;
        if (train.trainCategory === 'Long-distance') {
          number = train.trainType + ' ' + train.trainNumber;
        } else {
          number = train.commuterLineID + ' ' + train.trainNumber;
        }

        const estimatedTime =
          timeTableRow.liveEstimateTime === undefined
            ? ''
            : new Date(timeTableRow.liveEstimateTime);

        const origin = getStationName(
          train.timeTableRows[0].stationShortCode,
          stations,
        );

        const destination = getStationName(
          train.timeTableRows[train.timeTableRows.length - 1].stationShortCode,
          stations,
        );

        data.push({
          number,
          origin,
          destination,
          estimatedTime,
          type: train.trainType,
          time: scheduledTime,
          cancelled: timeTableRow.cancelled,
        });
      }
    });
  });
};
```